src/material-experimental/mdc-form-field/_form-field-high-contrast.scss

```diff
@@ -0,0 +1,36 @@
+@use '../../cdk/a11y';
+
+@mixin private-form-field-high-contrast() {
+  $focus-indicator-width: 3px;
+  $focus-indicator-style: dashed;
+
+  // The outline of the `fill` appearance is achieved through a background color
+  // which won't be visible in high contrast mode. Add an outline to replace it.
+  .mat-form-field-appearance-fill .mat-mdc-text-field-wrapper {
+    @include a11y.high-contrast(active, off) {
+      outline: solid 1px;
+    }
+  }
+
+  // If a form field with fill appearance is focused, update the outline to be
+  // dashed and thicker to indicate focus.
+  .mat-form-field-appearance-fill.mat-focused .mat-mdc-text-field-wrapper {
+    @include a11y.high-contrast(active, off) {
+      outline: $focus-indicator-style $focus-indicator-width;
```

Comment:
You can set `outline-style` instead of overriding the entire `outline`.

Reply:
Does that work if I also want to increase the width? I couldn't get it to work.

Reply:
Ah ok, so it goes from solid to dashed and it increases the width. IMO it'll look better if we kept it solid and just increased the width, similarly to how we do it on buttons.

Reply:
I'd argue that it's still difficult to figure out (on a page with a lot of form fields) which field is focused. The dashed border becomes more obvious in that case (subjectively)

Reply:
We should keep in mind that focus happens after a user interaction and the browser scrolls it into view. It's not like the user will have to look around for a focused element from the beginning.

Reply:
That's true; although I've somehow ran into that scenario where I briefly lost track which field is focused. i.e. when I started pressing tab somewhere.

I don't think keeping it dashed is really problematic. It's not like high-contrast is visually appealing anyway.

Reply:
I don't feel too strongly about it, but I'd go with either having the dashed one on top of the solid one, or replace the solid one with a dashed one (to make it look a little better; like I had initially) 

I think the dashed outline is visually easier to distinguish from the standard outline notch (which could be also thicker when the field is invalid = 2px for invalid fields)